mlflow/store/artifact/hdfs_artifact_repo.py

```diff
@@ -164,15 +164,13 @@ def hdfs_system(host, port):
     """
     import pyarrow as pa
 
-    driver = os.getenv('MLFLOW_HDFS_DRIVER') or 'libhdfs'
```

Comment:
can we still look if the environment is set and if it's set to something other than libhdfs? If that is the case, can we check the pyarrow version and raise an exception if it's incompatible?